### PR TITLE
must-gather:do not create  helper pod if storagecluster is deleted

### PR DESCRIPTION
**Pull request by rajatsing**
with storage cluster deleted, MG should not attempt to bring up helper pod.

Signed-off-by: RAJAT SINGH <rajasing@redhat.com>

### CODE REVIEW

**Review by crombus — CHANGES_REQUESTED** [review]
don't change the order. we install debug pods also. in pre-install script.
make changes in pre-install script to skip the deployment of helper pod

**Comment by rajatsing** [comment]
> don't change the order. we install debug pods also. in pre-install script.
> make changes in pre-install script to skip the deployment of helper pod

Sure, I have made changes in the `pre-install` script file.
Please have a look,it works as expected. I have tried.

File: must-gather/collection-scripts/pre-install.sh [hunk 1]
```diff
@@ -36,10 +39,12 @@ deploy(){
           fi
      fi
 
-     for node in ${nodes}; do
+    if [ "$storageClusterPresent" ]; then
+    for node in ${nodes}; do
```
Comment (crombus) [inline must-gather/collection-scripts/pre-install.sh:43]: the creation of debug pod is independent of storagecluster.

File: must-gather/collection-scripts/pre-install.sh [hunk 2]
```diff
@@ -65,12 +70,16 @@ check_for_helper_pod(){
...
 labels
 pids=()
-check_for_helper_pod &
+if [ "$storageClusterPresent" ]; then
+    check_for_helper_pod &
+fi
 pids+=($!)
-for node in ${nodes}; do
+if [ "$storageClusterPresent" ]; then
+    for node in ${nodes}; do
```
Comment (crombus) [inline must-gather/collection-scripts/pre-install.sh:78]: same here.

File: must-gather/collection-scripts/gather [hunk 3]
```diff
@@ -19,15 +20,17 @@ gather_clusterscoped_resources ${BASE_COLLECTION_PATH} "${SINCE_TIME}"
...
 else
     gather_ceph_resources ${BASE_COLLECTION_PATH} "${SINCE_TIME}"
 fi
 
-# Call post-uninstall.sh
-post-uninstall.sh
+if [ "${storageClusterPresent}" ]; then
+    # Call post-uninstall.sh
+    post-uninstall.sh
+fi
```
Comment (crombus) [inline must-gather/collection-scripts/gather:33]: this will also not delete debug pods. make the required changes in post-uninstall.sh script

**Review by crombus — CHANGES_REQUESTED** [review]
adding the logs to gather-debug.log commands should be in separate commit.

File: must-gather/collection-scripts/pre-install.sh (hunk 2 above)
Comment (rajatsing) [reply must-gather/collection-scripts/pre-install.sh]: fixed

File: must-gather/collection-scripts/gather (hunk 3 above)
Comment (rajatsing) [reply must-gather/collection-scripts/gather]: fixed

File: must-gather/collection-scripts/pre-install.sh (hunk 1 above)
Comment (rajatsing) [reply must-gather/collection-scripts/pre-install.sh]: fixed

File: must-gather/collection-scripts/pre-install.sh [hunk 4]
```diff
@@ -43,13 +48,15 @@ deploy(){
 }
 
 labels(){
+    if [ -n "${storageClusterPresent}" ]; then
```
Comment (crombus) [inline must-gather/collection-scripts/pre-install.sh:51]:
```suggestion
    if [ -z "${storageClusterPresent}" ]; then
```

File: must-gather/collection-scripts/pre-install.sh [hunk 5]
```diff
@@ -43,13 +48,15 @@ deploy(){
...
      oc label pod -n openshift-storage "${HOSTNAME}"-helper must-gather-helper-pod=''
+    fi
 }
 
 check_for_debug_pod(){
     debug_pod_name=$(oc get pods -n openshift-storage | grep "${node//./}-debug" | awk '{print $1}')
     # sleep for 60 seconds giving time for debug pod to get created
-    sleep 60 
+    sleep 60
```
Comment (crombus) [inline must-gather/collection-scripts/pre-install.sh:59]: It is good to have a clean code. I have one request can raise another pr fixing all the spaces in one commit.
Every-time Having separate commit for spaces is clumsy. 
Can we have a test for this so that spaces cannot be added in the first place.
for go code we have golangci check for this.

File: must-gather/collection-scripts/pre-install.sh [hunk 6]
```diff
@@ -65,8 +72,10 @@ check_for_helper_pod(){
 deploy
 labels
 pids=()
-check_for_helper_pod &
-pids+=($!)
+if [ -n "${storageClusterPresent}" ]; then
```
Comment (crombus) [inline must-gather/collection-scripts/pre-install.sh:75]: -z

File: must-gather/collection-scripts/gather [hunk 7]
```diff
@@ -18,10 +18,11 @@ gather_namespaced_resources ${BASE_COLLECTION_PATH} "${SINCE_TIME}"
...
 gather_noobaa_resources ${BASE_COLLECTION_PATH} "${SINCE_TIME}"
 
+storageClusterPresent=$(oc get storagecluster -n openshift-storage -o go-template='{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}')
 if [ "$(oc get storagecluster -n openshift-storage -o go-template='{{range .items}}{{.spec.externalStorage.enable}}{{"\n"}}{{end}}')" == true ]; then
-    echo "Skipping the ceph collection as External Storage is enabled"
-elif [ -z "$(oc get storagecluster -n openshift-storage -o go-template='{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}')" ]; then
-    echo "Skipping ceph collection as Storage Cluster is not present"
+    echo "Skipping the ceph collection as External Storage is enabled" | tee -a  "${BASE_COLLECTION_PATH}"/gather-debug.log
+elif [ -z "${storageClusterPresent}" ]; then
+    echo "Skipping ceph collection as Storage Cluster is not present" | tee -a  "${BASE_COLLECTION_PATH}"/gather-debug.log
```
Comment (crombus) [inline must-gather/collection-scripts/gather:25]: adding to the gather-debug.log should be in separate commit.

**Review by crombus — CHANGES_REQUESTED** [review]

File: must-gather/collection-scripts/pre-install.sh (hunk 4 above)
Comment (rajatsing) [reply must-gather/collection-scripts/pre-install.sh:51]: I would be `-n` because we want to execute that part when the storagecluster is present`-z`.
-n is the opposite of -z. `if [ -n "${storageClusterPresent}" ]` passes if the string is not null and not empty.

File: must-gather/collection-scripts/pre-install.sh (hunk 6 above)
Comment (rajatsing) [reply must-gather/collection-scripts/pre-install.sh:75]: https://stackoverflow.com/questions/18096670/what-does-z-mean-in-bash

File: must-gather/collection-scripts/gather (hunk 7 above)
Comment (rajatsing) [reply must-gather/collection-scripts/gather:25]: will do, I did not know that.

File: must-gather/collection-scripts/pre-install.sh (hunk 5 above)
Comment (rajatsing) [reply must-gather/collection-scripts/pre-install.sh:59]: Sure, we can get this merged and from the next time, we can use your fix for fixing empty spaces.

File: must-gather/collection-scripts/gather (hunk 7 above)
Comment (rajatsing) [reply must-gather/collection-scripts/gather]: done!.

**Review by crombus — APPROVED** [review]

**Comment by rajatsing** [comment]
Rebased the PR!

**Comment by crombus** [comment]
tested the changed.

/lgtm
